cmd/genfeaturegates/genfeaturegates.go

```diff
@@ -0,0 +1,598 @@
+		deps := allDependencies[featuregate.Feature(feature)]
+		if len(deps) > 0 {
+			depItems := make([]string, len(deps))
+			for i, d := range deps {
+				depItems[i] = string(d)
+			}
+			info.deps = strings.Join(depItems, "<br>")
+		}
+
+		info.linkCode = fmt.Sprintf("[code](https://cs.k8s.io/?q=%%5Cb%s%%5Cb&i=nope&files=&excludeFiles=CHANGELOG&repos=kubernetes/kubernetes)", feature)
+		info.linkKEPs = fmt.Sprintf("[KEPs](https://cs.k8s.io/?q=%%5Cb%s%%5Cb&i=nope&files=&excludeFiles=CHANGELOG&repos=kubernetes/enhancements)", feature)
+
+		features = append(features, info)
+	}
+
+	// Sort features based on sortBy parameter
+	sortFeatures(features, sortBy, reverseSort)
+
+	// Build markdown output
+	var sb strings.Builder
+	sb.WriteString("# Kubernetes Feature Gates\n\n")
+	sb.WriteString("<!-- This file is auto-generated by hack/update-featuregates.sh. DO NOT EDIT. -->\n")
+	sb.WriteString("<!-- To resolve merge conflicts, run: hack/update-featuregates.sh -->\n\n")
+
+	if filterStage != "" {
+		sb.WriteString(fmt.Sprintf("*Showing only %s features (%d)*\n\n", strings.ToUpper(filterStage), len(features)))
+	}
+
+	sb.WriteString("| Feature | Enabled | Locked | Alpha | Beta | GA | Deprecated | Dependencies | Links |\n")
+	sb.WriteString("|---------|---------|--------|-------|------|----|------------|--------------|-------|\n")
+
+	for _, info := range features {
+		sb.WriteString(fmt.Sprintf("| <nobr>%s</nobr> | <nobr>%s</nobr> | <nobr>%s</nobr> | <nobr>%s</nobr> | <nobr>%s</nobr> | <nobr>%s</nobr> | <nobr>%s</nobr> | <nobr>%s</nobr> | <nobr>%s %s</nobr> |\n",
+			info.name,
+			formatEnabledColumn(info.enabledFromVersion),
+			formatLockedColumn(info.lockedFromVersion),
+			formatStageRanges(info.alphaStages),
+			formatStageRanges(info.betaStages),
+			formatStageRanges(info.gaStages),
+			formatStageRanges(info.depStages),
+			info.deps, info.linkCode, info.linkKEPs))
+	}
+
+	sb.WriteString("\n## Legend\n\n")
+	sb.WriteString("| Symbol | Meaning |\n")
+	sb.WriteString("|--------|--------|\n")
+	sb.WriteString("| :ballot_box_with_check: | Enabled by default (in Enabled column: version when first enabled) |\n")
+	sb.WriteString("| :closed_lock_with_key: | Locked to default, cannot be changed (in Locked column: version when first locked) |\n")
+	sb.WriteString("| 1.X– | Feature is in this state from version 1.X (open-ended range) |\n")
+
+	return sb.String()
+}
+
+// formatEnabledColumn formats the Enabled column showing when a feature was first enabled by default
+func formatEnabledColumn(enabledFromVersion string) string {
+	if enabledFromVersion == "" {
+		return ""
+	}
+	return ":ballot_box_with_check: " + enabledFromVersion + "–"
+}
+
+// formatLockedColumn formats the Locked column showing when a feature was first locked
+func formatLockedColumn(lockedFromVersion string) string {
+	if lockedFromVersion == "" {
+		return ""
+	}
+	return ":closed_lock_with_key: " + lockedFromVersion + "–"
+}
```

Comment:
I think enabled and locked can probably stay `+` ... once enabled or locked, we *do* expect those to stay enabled/locked in future versions